eng/Versions.props

```diff
@@ -25,26 +25,20 @@
     <UsingToolVSSDK>true</UsingToolVSSDK>
   </PropertyGroup>
   <!-- Production Dependencies -->
-  <!-- Condition consumption of maintenance-packages dependencies based on source build.
-       This is to prevent "package downgrade" errors coming from other packages that are
-       already consuming the newest version of these same dependencies. -->
-  <PropertyGroup Condition="'$(DotNetBuildSourceOnly)' == 'true'">
+  <PropertyGroup>
+    <!-- manually maintained versions -->
     <MicrosoftIORedistVersion>6.1.3</MicrosoftIORedistVersion>
-    <SystemMemoryVersion>4.6.3</SystemMemoryVersion>
-    <SystemRuntimeCompilerServicesUnsafeVersion>6.1.2</SystemRuntimeCompilerServicesUnsafeVersion>
-    <SystemThreadingTasksExtensionsVersion>4.6.3</SystemThreadingTasksExtensionsVersion>
-  </PropertyGroup>
-  <PropertyGroup Condition="'$(DotNetBuildSourceOnly)' != 'true'">
```

Comment:
No more conditions anymore then?

Reply:
The condition is only necessary when the desktop msbuild inbox assemblies aren't on latest. This PR is bringing them to latest.

Reply:
Feels like we should maybe plan for both "a bunch of the time it can match" and also "a bunch of the time the netfx bits will need to match VS which is slightly stale". Maybe leave the conditions? Or have a new condition that encapsulates the existing one?

Reply:
I can duplicate the versions, yeah. I was debating with myself if that makes sense. But I was also hoping that we don't release a new version soon-ish.

Reply:
done